Plugins/EDM4hep/include/Acts/Plugins/EDM4hep/EDM4hepUtil.hpp

```diff
@@ -0,0 +1,420 @@
+  // [dtheta\sin(theta)/  dq/p\sin(theta)/]
+  //
+  // =
+  //
+  // [     2                        ]
+  // [- cot (theta) - 1       0     ]
+  // [                              ]
+  // [-B*q/p*cos(theta)       B     ]
+  // [------------------  ----------]
+  // [      2             sin(theta)]
+  // [   sin (theta)                ]
+
+  ActsSymMatrix<6> J;
+  J.setIdentity();
+  double cotTheta = std::tan(M_PI_2 + theta);
+  J(3, 3) = -cotTheta * cotTheta - 1;  // d(tanLambda) / dTheta
+  J(4, 4) = Bz / std::sin(theta);      // dOmega / d(qop)
+  double sinTheta = std::sin(theta);
+  J(4, 3) = -Bz * qOverP * std::cos(theta) /
+            (sinTheta * sinTheta);  // dOmega / dTheta
+  return J;
+}
+
+inline ActsSymMatrix<6> jacobianFromEdm4hep(double tanLambda, double omega,
+                                            double Bz) {
+  // [     d      /                     pi\                                  ]
+  // [------------|-atan(\tan\lambda) + --|                 0                ]
+  // [d\tan\lambda\                     2 /                                  ]
+  // [                                                                       ]
+  // [     d      /         \Omega        \     d   /         \Omega        \]
+  // [------------|-----------------------|  -------|-----------------------|]
+  // [d\tan\lambda|     __________________|  d\Omega|     __________________|]
+  // [            |    /            2     |         |    /            2     |]
+  // [            \B*\/  \tan\lambda  + 1 /         \B*\/  \tan\lambda  + 1 /]
+  //
+  // =
+  //
+  // [         -1                                     ]
+  // [   ----------------                 0           ]
+  // [              2                                 ]
+  // [   \tan\lambda  + 1                             ]
+  // [                                                ]
+  // [  -\Omega*\tan\lambda               1           ]
+  // [-----------------------  -----------------------]
+  // [                    3/2       __________________]
+  // [  /           2    \         /            2     ]
+  // [B*\\tan\lambda  + 1/     B*\/  \tan\lambda  + 1 ]
+
+  ActsSymMatrix<6> J;
+  J.setIdentity();
+  J(3, 3) = -1 / (tanLambda * tanLambda + 1);
+  J(4, 3) = -1 * omega * tanLambda /
+            (Bz * std::pow(tanLambda * tanLambda + 1, 3. / 2.));
+  J(4, 4) = 1 / (Bz * std::sqrt(tanLambda * tanLambda + 1));
+
+  return J;
+}
+
+inline void packCovariance(const ActsSymMatrix<6>& from, float* to) {
+  for (int i = 0; i < from.rows(); i++) {
+    for (int j = 0; j <= i; j++) {
+      size_t k = (i + 1) * i / 2 + j;
+      to[k] = from(i, j);
+    }
+  }
+}
+
+inline void unpackCovariance(const float* from, ActsSymMatrix<6>& to) {
+  auto k = [](size_t i, size_t j) { return (i + 1) * i / 2 + j; };
+  for (int i = 0; i < to.rows(); i++) {
+    for (int j = 0; j < to.cols(); j++) {
+      to(i, j) = from[j <= i ? k(i, j) : k(j, i)];
+    }
+  }
+}
```

Comment:
I wonder if eigen could do this for us?

Reply:
Not that I'm aware. If you find anything let me know!